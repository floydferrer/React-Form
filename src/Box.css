.Box {
    display: inline-block;
    box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.16), 0 3px 3px 0 rgba(0, 0, 0, 0.23);
    margin: 10px;
    vertical-align: middle;
}

.Box-text {
    opacity: 0;
}

.Box-text:hover {
    opacity: 1;
    cursor: pointer;
}