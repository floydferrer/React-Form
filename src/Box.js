import React from "react";
import './Box.css'

const Box = ({id, color, width, height, removeBox}) => {
    return (
        <div
            onClick={removeBox}
            role='box'
            className='Box'
            id={id}
            style={{
                backgroundColor: color,
                height: `${height}px`,
                width: `${width}px`,
            }}
        >
            <div
                className="Box-text"
                style={{
                    lineHeight: `${height}px`,
                    fontSize: `${height / 2}px`
                }}
            >
                X
            </div>
        </div>
    )
}

export default Box;