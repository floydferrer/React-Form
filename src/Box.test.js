import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", function() {
    render(<Box
      color='test-red'
      width='200'
      height='200'
    />)
});

it("matches snapshot", function() {
    const { asFragment } = render(<Box 
        color='test-red'
        width='200'
        height='200'
    />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("removes box when clicked on", function() {
    const removeBox = jest.fn();
    const { getByText, getByRole } = render(
      <Box
        color='test-red'
        width='200'
        height='200'
        removeBox={removeBox}
      />
    );
    const removeBtn = getByRole('box');
    fireEvent.click(removeBtn);

    expect(removeBox).toHaveBeenCalled();
  });