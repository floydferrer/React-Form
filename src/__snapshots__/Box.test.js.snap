// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    role="box"
    style="height: 200px; width: 200px;"
  >
    <div
      class="Box-text"
      style="line-height: 200px; font-size: 100px;"
    >
      X
    </div>
  </div>
</DocumentFragment>
`;
